src/js/fs.js

```diff
@@ -275,6 +275,76 @@ fs.readFileSync = function(path) {
 };
 
 
+fs.writeFile = function(path, data, callback) {
+  checkArgString(path);
+  checkArgBuffer(data);
+  checkArgFunction(callback);
+
+  var fd;
+  var len;
+  var bytesWritten;
+
+  fs.open(path, 'w', function(err, _fd) {
+    if (err) {
+      return callback(err);
+    }
+
+    fd = _fd;
+    len = data.length;
+    bytesWritten = 0;
+
+    write();
+  });
+
+  var write = function() {
+    fs.write(fd, data, bytesWritten, len - bytesWritten, 0, afterWrite);
+  };
+
+  var afterWrite = function(err, n) {
+    if (err) {
+      fs.close(fd, function(err) {
+        return callback(err);
+      });
+    }
+
+    if (n <= 0 || bytesWritten + n == len) {
+      // End of data
+      fs.close(fd, function(err) {
+        callback(err);
+      });
+    } else {
+      // continue writing
+      bytesWritten += n;
+      write();
+    }
+  };
+};
+
+
+fs.writeFileSync = function(path, data) {
+  checkArgString(path);
+  checkArgBuffer(data);
+
+  var fd = fs.openSync(path, 'w');
+  var len = data.length;
+  var bytesWritten = 0;
+
+  while (true) {
+    try {
+      var n = fs.writeSync(fd, data, bytesWritten, len - bytesWritten, 0);
```

Comment:
Not relevant, all locals are globals from function perspective. So never write:

    for (var i = 0; i < 5; i++) for (var i = 0; i < 5; i++) {}

Because it will not work as in C